pkg/controllers/vdb/promotesandboxtomain_reconciler.go

```diff
@@ -70,6 +69,9 @@ func (s *PromoteSandboxToMainReconciler) Reconcile(ctx context.Context, _ *ctrl.
 	}
 
 	sandboxName := s.PFacts.GetSandboxName()
+	if sandboxName == "" {
```

Comment:
`if sandboxName == vapi.MainCluster`

Also this should be before podfacts collection.